ambry-store/src/main/java/com/github/ambry/store/StorageManager.java

```diff
@@ -1002,6 +1009,33 @@ public void onPartitionBecomeDroppedFromOffline(String partitionName) {
       logger.info("Partition {} is successfully dropped on current node", partitionName);
     }
 
+
+
+    @Override
+    public void buildStateForFileCopy(String partitionName) {
+      // The partition map should have the replica for the current partition since it was called in pre-file-copy step.
+      ReplicaId replica = partitionNameToReplicaId.get(partitionName);
+
+      if (replica == null) {
+        logger.error("No existing replica found for partition {} in partitionNameToReplicaId", partitionName);
+        throw new StateTransitionException(
+            "Existing replica " + partitionName + " is not found in clustermap for " + currentNode, ReplicaNotFound);
+      }
+      Store store = getStore(replica.getPartitionId(), false);
+      try {
+        store.start();
+      } catch (StoreException e) {
+        logger.error("Failed to start new added store for partition id {}", replica.getPartitionId(), e);
+      }
+      DiskManager diskManager = partitionToDiskManager.get(replica.getPartitionId());
+      if (diskManager == null){
+        logger.error("DiskManager not found for partition {} in partitionToDiskManager", partitionName);
+        throw new StateTransitionException(
+            "Existing Disk manager not found for replica " + partitionName + " is not found in clustermap for " + currentNode, DiskManagerNotFoundForFileCopyStateBuild);
+      }
+      diskManager.addBlobStoreToCompactionManager((BlobStore) store);
```

Comment:
store.start() and diskManager.addBlobStoreToCompactionManager((BlobStore) store) can be moved to file copy manager class.

Reply:
Can you elaborate as to why would that be advantageous?